src/combinatorics/simple_graph/connectivity.lean

```diff
@@ -1109,6 +1131,19 @@ begin
       simpa using h.2, } },
 end
 
+/-- The specialization of `simple_graph.walk.map` for mapping walks to supergraphs. -/
+@[reducible] def map_le {G G' : simple_graph V} (h : G ≤ G') {u v : V} (p : G.walk u v) :
+  G'.walk u v := p.map (hom.map_spanning_subgraphs h)
+
+lemma is_trail.map_le {G G' : simple_graph V} (h : G ≤ G') {u v : V} {p : G.walk u v}
+  (pt : p.is_trail) : (p.map_le h).is_trail := map_is_trail_of_injective (function.injective_id) pt
```

Comment:
Now that we have the `iff` version, I suppose we can turn this into an iff and make it a simp lemma:

```lean
@[simp] lemma map_le_is_trail {G G' : simple_graph V} (h : G ≤ G') {u v : V} {p : G.walk u v} :
  (p.map_le h).is_trail ↔ p.is_trail :=
map_is_trail_iff_of_injective function.injective_id

alias map_le_is_trail ↔ is_trail.of_map_le is_trail.map_le
```

Could you do this for this one and the path and cycle versions too? The `alias` commands give the current versions, which could be useful with dot notation.